stateMutators.js

```diff
@@ -0,0 +1,174 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => {
+    return {
+        ...state,
+        basketPos: value,
+    }
+};
+
+const moveBasketLeft = (state, value) => {
+    return {
+        ...state,
+    basketPos: basketPos - value,
+    }
+};
+
+const moveBasketRight = (state, value) => {
+    return {
+        ...state,
+        basketPos: basketPos + value,
+    }
+};
+
+const setBasketValue = (state, value) => {
+    return {
+        ...state,
+        basketValue: value,
+    }
+};
+
+const resetBasketValue = (state) => {
+    return {
+        ...state,
+        basketValue: 0,
+    }
+};
+
+const setScore = (state, value) => {
+    return {
+        ...state,
+        score: value,
+    }
+};
+
+const resetScore = (state) => {
+    return {
+        ...state,
+        score: 0,
+    }
+};
+
+const loseLife = (state, value) => {
+    return {
+        ...state,
+        livesRemaining: value,
+    }
+};
+
+const resetLivesRemaining = (state) => {
+    return {
+        ...state,
+        livesRemaining: 3,
+    }
+};
+
+const setGameLevel = (state, value) => {
+    return {
+        ...state,
+        gameLevel: value,
+    }
+};
+
+const resetGameLevel = (state) => {
+    return {
+        ...state,
+        gameLevel: 0,
+    }
+};
+
+const setGameMode = (state, value) => {
+    return {
+        ...state,
+        gameMode: value,
+    }
+};
+
+const addFallingObject = (state, fallingObj) => {
+    return {
+        ...state,
+        fallingObjects: [
+            ...state.fallingObjects,
+            fallingObj,
+        ]
+    }
+};
+
+const removeFallingObject = (state, fallingObjId) => {
+    return {
+        ...state,
+        fallingObjects: [
+            ...state.fallingObjects.slice[0, fallingObjId],
```

Comment:
`slice` is a function on `Array.prototype` so you need to use round brackets to call it. Square are for array or object index access. i.e., `.slice(0, fallingObjId)`

Reply:
@timotiousPrime This got marked as resolved but it's still a problem in the code. Let me know if it's not clear

Reply:
Sorry, I completely missed that I used the wrong brackets. I have updated it now